gitlab/client.py

```diff
@@ -98,7 +108,9 @@ def __init__(
         self._set_auth_info()
 
         #: Create a session object for requests
-        self.session = session or requests.Session()
+        http_backend = kwargs.pop("http_backend", http_backends.DefaultBackend)
```

Comment:
Based on the comment above, this should disambiguate the session object as well as any other potential backend-related typing (needs a `Type` import at the top):

```suggestion
        http_backend: Type[http_backends.DefaultBackend] = kwargs.pop(
            "http_backend", http_backends.DefaultBackend
        )
```

Reply:
Sorry for my ignorance. Not that I mind adding this change but which concern is this change addressing for the current change? I would prefer to delay it to a later stage if that is related to the follow up changes.